AirCasting/SessionViews/ABMeasurementsView.swift

```diff
@@ -3,27 +3,51 @@
 
 import SwiftUI
 import AirCastingStyling
-import CoreLocation
-import CoreData
 
 enum MeasurementPresentationStyle {
     case showValues
     case hideValues
 }
 
 struct ABMeasurementsView: View {
+    var viewModelProvider: () -> CardMeasurementsViewModel
     @ObservedObject var session: SessionEntity
     @Binding var isCollapsed: Bool
     @Binding var selectedStream: MeasurementStreamEntity?
-    @Binding var showLoadingIndicator: Bool
     var thresholds: [SensorThreshold]
     let measurementPresentationStyle: MeasurementPresentationStyle
-    let sessionDownloader = SessionDownloadService(client: URLSession.shared,
-                                                   authorization: UserAuthenticationSession(),
-                                                   responseValidator: DefaultHTTPResponseValidator())
-    let measurementStreamStorage: MeasurementStreamStorage
-    @EnvironmentObject var selectedSection: SelectSection
 
+    @State private var viewModel: CardMeasurementsViewModel?
```

Comment:
Try to use `@StateObject` instead of `@State` (ah, SwiftUI...) - it will probably remove the need of `viewModelProvider` (you will just inject ViewModel in init as you would typically do)

Reply:
I might be wrong but I think general advice is that `@StateObject` should be private and not passed from the parent view.
I couldn't find much information about that. But here's one comment from Apple's engineer:
https://developer.apple.com/forums/thread/650618
From what I understand, he confirms that `@StateObject` should be initialized inside the view.
This I think makes sense. Even if passing a ViewModel to the @StateObjct wouldn't override the old object, it would still create that view model on each refresh or parent's view and deinitialize it immediately. It just smells bad 🤷🏻‍♀️